### PR TITLE
agent-ctl: Install agent-ctl in install_runtime.sh

### LINKED ISSUE #4602: Install agent-ctl on CI
Currently `agent-ctl` is not installed on CI environments. However, the tool can be used on tests to determine conditions, for example, whether the agent support seccomp before running a seccomp test (see https://github.com/kata-containers/tests/issues/4303). With the emerging confidential containers specific features (see https://github.com/kata-containers/ci/pull/455#issuecomment-1072413330), it is likely more tests will be relying on `agent-ctl`. Therefore, let's build and install it on CI.

### PR DESCRIPTION
**Pull request by bookinabox**
Installs agent-ctl in install_runtime.sh which allows better debugging
in the ci environment.

Fixes #4602

Signed-off-by: Derek Lee <derlee@redhat.com>

### CODE REVIEW

File: .ci/install_runtime.sh [hunk 1]
```diff
@@ -63,6 +66,19 @@ build_install_shim_v2(){
 
 build_install_shim_v2
 
+build_install_agent_ctl(){
+	if [ ! -d "$agent_ctl_path" ]; then
```
Comment (wainersm) [inline .ci/install_runtime.sh:70]: You should use the `clone_katacontainers_repo()`  method from `.ci/lib.sh` (already sourced). Then use the already exported `katacontainers_repo_dir` variable to reference the path to the kata-containers repo.

File: .ci/install_runtime.sh [hunk 2]
```diff
@@ -12,6 +12,8 @@ set -o errtrace
 
 cidir=$(dirname "$0")
 
+bash "${cidir}/install_rust.sh" && source "$HOME/.cargo/env"
```
Comment (wainersm) [inline .ci/install_runtime.sh:15]: I would call the rust install script from `build_install_agent_ctl()`.  Here it is loosely in between variables declarations...

File: .ci/install_runtime.sh [hunk 3]
```diff
@@ -46,6 +48,7 @@ OPENSHIFT_CI="${OPENSHIFT_CI:-false}"
 
 runtime_config_path="${SYSCONFDIR}/kata-containers/configuration.toml"
 runtime_src_path="${GOPATH}/src/${KATACONTAINERS_REPO}/src/runtime"
+agent_ctl_path="${GOPATH}/src/${KATACONTAINERS_REPO}/src/tools/agent-ctl"
```
Comment (wainersm) [inline .ci/install_runtime.sh:51]: use `katacontainers_repo_dir` variable. The `runtime_src_path` and other variables on this file are doing wrong.

**Comment by wainersm** [comment]
Thanks for working on this @bookinabox ! I left just a few comments.

**Comment by wainersm** [comment]
All the other CI jobs except for jenkins-ci-ppc64le-ubuntu-20-04 are passing, which is good. I remember @fidencio commenting somewhere that job was unstable recently. Let me re-trigger it.

**Review by wainersm — APPROVED** [review]

File: .ci/install_runtime.sh [hunk 4]
```diff
@@ -63,6 +62,17 @@ build_install_shim_v2(){
 
 build_install_shim_v2
 
+build_install_agent_ctl(){
+	bash "${cidir}/install_rust.sh" && source "$HOME/.cargo/env"
```
Comment (fidencio) [inline .ci/install_runtime.sh:66]: AFAIR we're not building any rust component on the host side, but rather building them inside a container.
Would be possible to do the same here?

Comment (bookinabox) [reply .ci/install_runtime.sh:66]: I'm not super familiar with the project still, and I'm a bit confused. the agent-ctl runs on the host side, right?

Comment (fidencio) [reply .ci/install_runtime.sh:66]: @bookinabox, oops, my bad, let me take a step back here.

It runs on the host side, but what I was afraid of was this being the only reason we end up installing rust in the CI machines, just to build the `agent-ctl` tool.  If that was the case, I'd suggest we rather build it inside a container (which would have rust installed there) and just install the binary in the correct location.

However, it seems we're already installing rust on the Ci machine as we build `virtiofsd` there (for non-x86_64 machines), thus I think this one is fine to be merged as it's.

Does the explanation make sense?

**Review by fidencio — APPROVED** [review]
lgtm, thanks @bookinabox!

File: .ci/install_runtime.sh (hunk 4 above)
Comment (bookinabox) [reply .ci/install_runtime.sh:66]: yes, thanks!

**Pull request merged**